src/main/java/frc/robot/util/ShuffleBoard/Tabs/SubsystemTab.java

```diff
@@ -1,20 +1,27 @@
 
 package frc.robot.util.ShuffleBoard.Tabs;
 
+import java.awt.Robot;
+
+import edu.wpi.first.wpilibj.drive.RobotDriveBase;
 import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import frc.robot.commands.TurnyCommand;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.subsystems.Ting;
 import frc.robot.util.ShuffleBoard.ShuffleBoardTabs;
 
 
 public class SubsystemTab extends ShuffleBoardTabs {
-    // TODO 2.3.11: Create variable for subsystem
+    Ting ting;
 
-    public SubsystemTab(){
-        
+    public SubsystemTab(Ting tingy){
+        ting = tingy;
     }
 
     public void createEntries(){
         tab = Shuffleboard.getTab("Subsystem");
 
+        tab.add("Ting", ting.getMechanism2d());
         // TODO 2.4.7: Add Mechanism2d
 
         // TODO 3.3.13: Add command buttons
```

Comment:
You still need to do 3.3.12 and 3.3.13.

Reply:
I did it, I just added it in Drivetab.java because I asked someone and they said since it requires the drivetrain

Reply:
That section involves making a Bang-Bang command for your subsystem. One if the steps said to make a command for your subsystem, and then 3.3.12 says to extend that command.